ethcore/light/src/on_demand/mod.rs

```diff
@@ -54,13 +54,13 @@ struct Peer {
 }
 
 impl Peer {
-	// whether this peer can fulfill the
+	// whether this peer can fulfill the necessary capabilities
 	fn can_fulfill(&self, c: &Capabilities) -> bool {
 		let caps = &self.capabilities;
 
-		caps.serve_headers == c.serve_headers &&
-			caps.serve_chain_since >= c.serve_chain_since &&
-			caps.serve_state_since >= c.serve_chain_since
+		caps.serve_headers >= c.serve_headers &&
```

Comment:
the purpose of this function is to check whether the capabilities `c` are encompassed by the capabilities `caps`. If `c.serve_headers` is `false`, `caps.serve_headers` can be `false` or `true. If `c.serve_headers` is `true`, `caps.serve_headers` can only be `true`.